src/MLJInterface.jl

```diff
@@ -84,20 +92,19 @@ MLJModelInterface.@mlj_model mutable struct LGBMRegressor <: MLJModelInterface.D
     is_provide_training_metric::Bool = false
     eval_at::Vector{Int} = Vector{Int}([1, 2, 3, 4, 5])::(all(_ .> 0))
 
-    # Implementation parameters
+    # Network parameters
     num_machines::Int = 1::(_ > 0)
-    num_threads::Int  = 0::(_ >= 0)
     local_listen_port::Int = 12400::(_ > 0)
     time_out::Int = 120::(_ > 0)
     machine_list_filename::String = ""
-    save_binary::Bool = false
```

Comment:
`save_binary` is the only param I removed in this refactoring from all estimators as it's a CLI param: https://lightgbm.readthedocs.io/en/v3.3.5/Parameters.html#save_binary